### PR TITLE
Remove default make target and added help

### PR DESCRIPTION
**Pull request by sajayantony**
Avoids accidentally running delete on current cluster. 

image

### CODE REVIEW

**Comment by sajayantony** [comment]
Can rebase after #4 if that helps.

**Comment by vsoch** [comment]
Yes was just going to suggest that! Sorry for the delayed push - I had a meeting the last hour and wanted to get it in before that (but ran out of time). I was able to fix the current bug with cleaning up the volume (deleting the pod) and the Makefile command that I had added for myself (and really should have removed). In addition I'm going to add a helm chart that will allow to control the image deployed (production vs. dev) so I anticipated that and added both of `build-dev` and `push-dev`.

File: Makefile [hunk 1]
```diff
@@ -4,16 +4,20 @@ DEVTAG=$(DRIVER_VERSION)-dev
...
+.PHONY: help
+help: ## Generates help for all targets
+	@grep -E '^[^#[:space:]].*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-uninstall:
+ready: clean compile ## Clean pods and compiles the driver
+publish-dev: clean compile build-dev push-dev  ## TODO 
+redo: uninstall compile build push install ## Single command to uninstall package and push
+
+uninstall: ## Unisntalls the plugin from the cluster
```
Comment (vsoch) [inline Makefile]:
```suggestion
uninstall: ## Uninstalls the plugin from the cluster
```

File: Makefile [hunk 2]
```diff
@@ -4,16 +4,20 @@ DEVTAG=$(DRIVER_VERSION)-dev
...
-ready: clean compile
-publish: uninstall compile build push install
-dev: uninstall compile build-dev push-dev install
+.PHONY: help
+help: ## Generates help for all targets
+	@grep -E '^[^#[:space:]].*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
-uninstall:
+ready: clean compile ## Clean pods and compiles the driver
+publish-dev: clean compile build-dev push-dev  ## TODO
```
Comment (vsoch) [inline Makefile]:
```suggestion
publish-dev: clean compile build-dev push-dev  ## Rebuild and publish development image
```
I'm good to rename `dev` to `publish-dev` but we will need to update that one reference to `make dev` in the README to `make publish-dev`. And if this is something we want to regularly run to develop, do you think `make dev` is not descriptive enough? I'm good with either, let's just make sure it's the same in both places.

File: Makefile [hunk 3]
```diff
@@ -4,16 +4,20 @@ DEVTAG=$(DRIVER_VERSION)-dev
 NAME=oras-csi-plugin
 DOCKER_REGISTRY=ghcr.io/converged-computing
 
-ready: clean compile
-publish: uninstall compile build push install
-dev: uninstall compile build-dev push-dev install
+.PHONY: help
+help: ## Generates help for all targets
+	@grep -E '^[^#[:space:]].*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
```
Comment (vsoch) [inline Makefile:9]: I love this! Looks great - I should add help commands like this to all my Makefile.

image

File: Makefile [hunk 4]
```diff
@@ -22,32 +26,32 @@ compile:
 	@echo "==> Building the project"
 	@env CGO_ENABLED=0 GOCACHE=/tmp/go-cache GOOS=linux GOARCH=amd64 go build -a -o cmd/oras-csi-plugin/${NAME} cmd/oras-csi-plugin/main.go
 
-push-dev: build-dev
+push-dev: build-dev ## Build and push images to devtag
```
Comment (vsoch) [inline Makefile]: To be consistent with the next one.
```suggestion
push-dev: build-dev ## Build and push images to DEVTAG
```

File: Makefile [hunk 5]
```diff
@@ -22,32 +26,32 @@ compile:
...
 	@docker build -t $(DOCKER_REGISTRY)/oras-csi-plugin:latest cmd/oras-csi-plugin
 
-build-dev: compile
+build-dev: compile ## Build images for DEVTAG
 	@echo "==> Building DEV docker images"
 	@docker build -t $(DOCKER_REGISTRY)/oras-csi-plugin:$(DEVTAG) cmd/oras-csi-plugin
 	@docker build -t $(DOCKER_REGISTRY)/oras-csi-plugin:latest cmd/oras-csi-plugin
 
-clean:
+clean: ## Deletes driver and all pods
```
Comment (vsoch) [inline Makefile]:
```suggestion
clean: ## Deletes driver
```
Thankfully I removed the "and all pods" part :)

**Comment by sajayantony** [comment]
Updated - 
image

**Review by vsoch — APPROVED** [review]
Perfecto! Thank you!

**Comment by vsoch** [comment]
I'll start (likely this evening) a small docs folder (that eventually can be more substantial) but to start will be Markdown, and can write up some of the ideas / use cases in more detail.

**Pull request merged**